src/output.h

```diff
@@ -208,4 +208,13 @@ void OutputLoggerExitPrintStats(ThreadVars *, void *);
 void OutputSetupActiveLoggers(void);
 void OutputClearActiveLoggers(void);
 
+typedef bool (*SimpleTxLogFunc)(void *, struct JsonBuilder *);
+
+typedef struct AppLayerLogger {
+    AppProto proto;
+    SimpleTxLogFunc LogTx;
+} AppLayerLogger;
```

Comment:
From a code organization perspective, I'd like to see `output.[ch]` just deal with the core/raw logging registration, which JSON/EVE should not be a part of.  So this looks a bit more like a `JsonAppLayerLogger`, and `SimpleTxLogFunc` looks more like a `SimpleJsonTxLogFunc`, or maybe simply `JsonTxLogFunc`?

Reply:
Ok for renaming.

`SimpleJsonTxLogFunc` does not cover all the cases (when the state is needed, to when some option flags are passed)